sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/rowBasedHiveUDFs.scala

```diff
@@ -55,6 +55,33 @@ trait GpuRowBasedHiveUDFBase extends GpuRowBasedUserDefinedFunction with HiveIns
 
   override def prettyName: String = name
 
+  /** Handle GpuLiteral and foldable GpuExpression, similar to `toInspector(Expression)` */
+  protected def gpuToInspector(expr: Expression): ObjectInspector = expr match {
+    case GpuLiteral(value, dataType) =>
+      // Convert to CPU literal as possible as we can before going into `toInspector(Expression)`.
+      // Because some optimization will be made for Literal expressions of primitive types.
```

Comment:
It would be nice to elaborate here what the issue is and why we need to do this.  "Some optimizations will be made" isn't clear why we need to worry and do anything special for literals.  If this code ends up causing problems in the future, it would be nice for those future maintainers to understand why this code was written in the first place.

Reply:
Updated.